smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CommandGenerator.java

```diff
@@ -474,7 +484,29 @@ private void addCommandSpecificPlugins() {
                 }
                 writer.pushState();
                 writer.putContext(symbolMap);
-                writer.write("$pluginFn:T(config" + additionalParamsString + "),");
+                writer.openBlock("$pluginFn:T(config", "),", () -> {
+                    List<String> additionalParameters = CodegenUtils.getFunctionParametersList(paramsMap);
+                    Map<String, CommandWriterConsumer> clientAddParamsWriterConsumers =
+                        plugin.getOperationAddParamsWriterConsumers();
+                    if (additionalParameters.isEmpty() && clientAddParamsWriterConsumers.isEmpty()) {
+                        return;
+                    }
+                    writer.writeInline(", { ");
+                    writer.writeInline(String.join(", ", additionalParameters));
+                    clientAddParamsWriterConsumers.forEach((key, consumer) -> {
+                        writer.writeInline(key).writeInline(": ");
```

Comment:
consider changing this line to use a consumer parameters. For example: 
```
writer.writerInline("$L: $C, ", key, w -> consumer.accept(w, ...))
```
I find this construction a bit easier to reason about.